cli/main.c

```diff
@@ -936,12 +982,43 @@ static int log_dump(int argc, char **argv)
 		{"type", 't', "TYPE", CFG_CHOICES, &cfg.type,
 		  required_argument,
 		 "log type to dump", .choices=types},
+		{"format", 'f', "FORMAT", CFG_CHOICES, &cfg.format,
+		  required_argument,
+		 "output log file format", .choices=format},
 		{NULL}};
 
 	argconfig_parse(argc, argv, CMD_DESC_LOG_DUMP, opts, &cfg, sizeof(cfg));
 
+	ret = switchtec_get_device_info(cfg.dev, &boot_phase, NULL, NULL);
+	if (ret) {
+		switchtec_perror("log_dump");
+		return ret;
+	}
+
+	if (boot_phase != SWITCHTEC_BOOT_PHASE_FW &&
+	    (cfg.type == SWITCHTEC_LOG_RAM ||
+	     cfg.type == SWITCHTEC_LOG_FLASH) &&
+	    cfg.format == LOG_FMT_TXT &&
+	    cfg.log_def_file == NULL) {
+		fprintf(stderr, "Cannot generate text format log file in BL1/2 boot phase without\n"
+				"a log defintion file. Please provide log definiton file with '-d',\n"
+				"or specify binary log format with '-f BIN' instead\n");
+		return -1;
+	}
+
+	if (cfg.format == LOG_FMT_BIN) {
+		log_def_to_use = NULL;
+	} else {
+		log_def_to_use = cfg.log_def_file;
+
+		ret = get_log_def_file(cfg.dev, cfg.type, cfg.format,
+				       &log_def_to_use);
+		if (ret < 0)
+			return ret;
+	}
```

Comment:
I feel like this would be slightly less awkward if written like this:

```c
if (cfg.format == LOG_FMT_BIN) {
    log_def_to_use = NULL; 
} else if (cfg.log_def_file) {
    log_def_to_use = cfg.log_def_file;
} else {
    log_def_to_use = get_log_def_file(...);
    if (!log_def_to_use)
          return -1;
}
```

Reply:
Good idea. 

Also added `INFO` print to notify user of an incorrect usage of `-f BIN` for other log types.